Framework/Algorithms/src/AbsorptionCorrection.cpp

```diff
@@ -323,17 +385,30 @@ void AbsorptionCorrection::retrieveBaseProperties() {
 /// Create the sample object using the Geometry classes, or use the existing one
 void AbsorptionCorrection::constructSample(API::Sample &sample) {
   const std::string xmlstring = sampleXML();
+  const std::string scatterFrom = getProperty("ScatterFrom");
   if (xmlstring.empty()) {
-    // This means that we should use the shape already defined on the sample.
-    m_sampleObject = &sample.getShape();
+    // Get the shape from the proper object
+    if (scatterFrom == CALC_SAMPLE)
+      m_sampleObject = &sample.getShape();
+    else if (scatterFrom == CALC_CONTAINER)
+      m_sampleObject = &(sample.getEnvironment().getContainer());
+    else if (scatterFrom == CALC_ENVIRONMENT)
+      m_sampleObject = &(sample.getEnvironment().getComponent(1));
+    else
+      throw std::runtime_error("Somebody forgot to fill in an if/else tree");
```

Comment:
Maybe use an `assert` here instead as it logically shouldn't get here?

Reply:
`assert` only functions in debug builds. I was trying to give a clear (enough) error message in the off chance that a user encountered it.

Reply:
Ok. I think maybe a slightly more clear help message at least indicating the parameter that is unexpected?